_examples/hi/test.py

```diff
@@ -168,100 +168,107 @@
 c.P2.Age = 51
 print(c)
 
-## test Couple.__init__
+# test Couple.__init__
 print("--- Couple.__init__")
 # Note: pybindgen does not automatically support varargs, so in general
 # all python calls need to provide the full Go signature of args.
 #c = hi.Couple(hi.Person("p1", 42))
-#print(c)
+# print(c)
 c = hi.Couple(hi.Person("p1", 42), hi.Person("p2", 52))
 print(c)
 c = hi.Couple(P1=hi.Person("p1", 42), P2=hi.Person("p2", 52))
 print(c)
 c = hi.Couple(P2=hi.Person("p1", 42), P1=hi.Person("p2", 52))
 print(c)
 
+# It's dangerous to refer to go handles as constants now that they
+# are gc'ed.
+anon1 = hi.Anon()
+h1 = anon1.handle
+h2 = anon.handle
+
 try:
-    hi.Couple(1)
+    hi.Couple(h1)
```

Comment:
I don't understand this change.
originally, this was testing type-correctness.

Reply:
well... this must be some uncaught regression.
(I haven't bisected but I suspect that's since the migration to `pybindgen`)
it used to be that invalid arguments would be caught and an exception be raised.
(I agree, printing `*ERROR* isn't a very robust mechanism...)

Reply:
so, to circle back on this.
I'd rather have the original test being commented out (with the `hi.Couple(1)` left as is) instead of changing the purpose of the test.

I have filed https://github.com/go-python/gopy/issues/217 so we don't forget about this.

Reply:
yep, that's my plan too. I looked back through the git history and can see what happened, there's a TODO in the code. I'll hopefully get this cleaned up today or tomorrow.

Reply:
ok, please take another look, especially at the comments in gen_struct. I think those changes make sense, but the underlying problem is that the confusion between int values that are handles and those that are not and I don't see any clean/easy way of resolving that beyond what I've done here. Let me know if you have any other ideas.